pkgs/development/libraries/boost/generic.nix

```
@@ -124,15 +124,20 @@ stdenv.mkDerivation {
     extraPrefix = "libs/context/";
   })
   ++ optional (and (versionAtLeast version "1.70") (!versionAtLeast version "1.73")) ./cmake-paths.patch
-  ++ optional (versionAtLeast version "1.73") ./cmake-paths-173.patch;
+  ++ optional (versionAtLeast version "1.73") ./cmake-paths-173.patch
+  ++ optional (stdenv.hostPlatform != stdenv.buildPlatform) (
```

```suggestion
  ++ (
```

I'm of the opinion that these patches should be applied unconditionally - see https://github.com/NixOS/nixpkgs/pull/105478#discussion_r579573484. This is especially true for boost, which already has a lot of potentially conflicting downstream patches.

---

@r-burns 
It leads to 10538 packages updates. Maybe we should merge this to `staging`? I'm not sure how to change "target branch" for an opened PR.

---

Yep staging for sure. If you hit the 'edit' button up top it'll let you change the target branch.